packages/router/router.ts

```diff
@@ -1915,8 +1917,12 @@ export function createRouter(init: RouterInit): Router {
       "Expected a location on the redirect navigation"
     );
 
-    // Check if this an external redirect that goes to a new origin
-    if (isBrowser && typeof window?.location !== "undefined") {
+    // Check if this an absolute external redirect that goes to a new origin
+    if (
+      ABSOLUTE_URL_REGEX.test(redirect.location) &&
+      isBrowser &&
+      typeof window?.location !== "undefined"
+    ) {
```

Comment:
This was a bit too aggressive - we only need to bother looking for a different origin if the redirect was an absolute URL.